src/textual/screen.py

```diff
@@ -120,14 +123,36 @@ def on_idle(self, event: events.Idle) -> None:
         elif self._dirty_widgets:
             self.update_timer.resume()
 
+        # The Screen is idle - a good opportunity to invoke the scheduled callbacks
+        await self._invoke_and_clear_callbacks()
+
     def _on_update(self) -> None:
         """Called by the _update_timer."""
         # Render widgets together
         if self._dirty_widgets:
             self._compositor.update_widgets(self._dirty_widgets)
             self.app._display(self._compositor.render())
             self._dirty_widgets.clear()
+
         self.update_timer.pause()
+        self.post_message_no_wait(events.PostScreenUpdate(self))
```

Comment:
Since we can't easily just make this method async without the function colouring problem arising, we make use of the queue to schedule execution of the async "invoke callbacks" functionality.

Reply:
Done, and renamed it to match this new behaviour.